### ISSUE TITLE
9.0 DOCS: Node `hidden` vs `disabled`

### ISSUE THREAD

**Issue by mhsdesign** [neos/neos-development-collection#4312]
Previously in neos 8.3 you could _hide_ a node. This is now called disabled (see command)

@bwaidelich described it as:

> What we sometimes call “hide” is actually a disable (and that’s how the event is called with Neos 9 and IMO we should reflect that in the UI as well).
I could even imagine, hiding (pun intended) disabled nodes from the document tree by default and adding some kind of “recycle bin” view… But I’m getting side-tracked: My point is that we simply can’t know whether the user…

I would like to have the differentiation and mind shift documented (maybe in the php doc codes) so people can understand ^^

the `VisibilityConstraints` seem to need docs in general:
https://github.com/neos/neos-development-collection/blob/ee36cc29460aa2005a22b577080a85ba505a6d1f/Neos.ContentRepository.Core/Classes/Projection/ContentGraph/VisibilityConstraints.php#L24

or maybe here in the command? https://github.com/neos/neos-development-collection/blob/4c9dd087324317c8b2823b2ee4a658b23d638892/Neos.ContentRepository.Core/Classes/Feature/NodeDisabling/Command/DisableNodeAggregate.php#L31

The general problem might be that integrators are familiar with the term `hidden` in neos but not with `disabled` im now conflicted myself how to name the property in the nodeType templates package ^^

So any definition would be welcome ;)

**Comment by bwaidelich** [comment]
The good old "hiddenInIndex" makes sense => The page won't appear in Menus, but it can still be navigated to.
"hidden" didn't really make sense, because it's not that the page is just hidden and you can find it if you only have the correct URL. That node does not exist (as far as the frontend is concerned) but it can be _reactivated_.

A bit similar to files you put into the recycler (vs files you _hide_ from the file explorer/finder)

**Comment by Sebobo** [comment]
IMO we have to be a bit careful with naming and where we use "technically" correct terms and where we use terms that our users actually understand.
Whenever I read CR related technical text, I'm getting quite confused and have trouble "translating" it to the people I help on a daily basis.

Do we have some glossary that we can review and discuss at some point?

**Comment by mhsdesign** [comment]
Im not sure i get how it all relates together. My brain could manage `VisibilityConstraints` and `DisableNodeAggregate` but now im a bit confused by: `NodeHiddenState` (especially why is it `hidden`, and why `can it NOT answer the question whether a Node is hidden because some node above it has been hidden`?)

**Comment by bwaidelich** [comment]
> IMO we have to be a bit careful with naming and where we use "technically" correct terms
> and where we use terms that our users actually understand.

I fully agree and I also stumbled upon the many new names and concepts in the new CR.

> Do we have some glossary that we can review and discuss at some point?

I don't think so, but we urgently need that on docs.neos.io – I created an issue for that and put it on the project board (https://github.com/neos/neos-development-collection/issues/4314)

> im a bit confused by: NodeHiddenState

Yes, that name is a left-over and it should be named `NodeDisabledState` – or better yet, be extended to all kinds of attributes (see #4315

**Comment by mhsdesign** [comment]
I just found this on docs neos io

https://docs.neos.io/guide/contributing-to-neos/event-sourced-content-repository/enabled-and-disabled-nodes


> In the old CR, this was called “hidden”, but we felt that enabled and disabled captured the intent better.
>
> If a node is disabled, the system should behave as if this node would not exist. This means:
>
> - you are not able to query for this node
> - you are not able to query for any child node: If you removed the node, all children would be removed as well.
> - these rules must only apply in frontend, when viewing the site. In backend, you should still see the disabled nodes, so that you can e.g. enable them again.

**Comment by mhsdesign** [comment]
Also this is the pr fyi which renamed it to `disabled` ^^

https://github.com/neos/contentrepository-development-collection/pull/94

**Comment by mhsdesign** [comment]
@dlubitz and me just wondered how far we would like to take this naming decision. Eg should this also be reflected in labels of the Neos Ui

https://github.com/neos/neos-development-collection/pull/4817#discussion_r1472773434

**Comment by mhsdesign** [comment]
@bwaidelich @kitsunet @grebaldi and me discussed to align the naming also in the Ui to "Disabled" (at least in english) In german "Deaktivieren" "Ausschalten" ...
